src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/NamingStyle.cs

```diff
@@ -179,7 +179,7 @@ public bool IsNameCompliant(string name, out string failureReason)
             // remove specified Prefix, then look for any other common prefixes
             name = StripCommonPrefixes(name.Substring(Prefix.Length), out var prefix);
 
-            if (prefix != string.Empty)
+            if (prefix != string.Empty && !char.IsDigit(name[0]))
```

Comment:
not sure if this is ok.  i.e. if would thinkg `_0foo` was ok.  i think it should check all the remainder of hte string and only ignore if it's all a number.

Reply:
would also need a comment explaining this.

Reply:
Perhaps this helper method from Remove Unused Parameter could be moved so it can be re-used?
https://github.com/dotnet/roslyn/blob/master/src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs#L327-L334

This allows for only one digit after an underscore, but at least we'd have consistency.

Reply:
Thanks @davidwengier resolving the conflicting nature of the two rules was my goal with #47569